lib/APINotes/APINotesYAMLCompiler.cpp

```diff
@@ -370,6 +385,30 @@ namespace llvm {
       }
     };
 
+    template<>
+    struct ScalarEnumerationTraits<api_notes::EnumExtensibilityKind> {
+      static void enumeration(IO &io, api_notes::EnumExtensibilityKind &value) {
+        io.enumCase(value, "none",   api_notes::EnumExtensibilityKind::None);
+        io.enumCase(value, "open",   api_notes::EnumExtensibilityKind::Open);
+        io.enumCase(value, "closed", api_notes::EnumExtensibilityKind::Closed);
+      }
+    };
+
+    template<>
+    struct ScalarEnumerationTraits<EnumConvenienceAliasKind> {
+      static void enumeration(IO &io, EnumConvenienceAliasKind &value) {
```

Comment:
Syntactic sugar for API notes! I guess this breaks round-tripping through the binary API notes format, but whatever; it certainly makes it easier for authors of API notes files to get things right.